File: src/Java/Annotation/TextOrigin.java
import vtk.vtkNativeLibrary;
import vtk.vtkActor;
import vtk.vtkAxes;
import vtk.vtkVectorText;
import vtk.vtkPolyDataMapper;
import vtk.vtkFollower;
import vtk.vtkNamedColors;

import vtk.vtkRenderWindow;
import vtk.vtkRenderWindowInteractor;
import vtk.vtkRenderer;


public class TextOrigin  {
	 // private static final long serialVersionUID = 1L;
	 // -----------------------------------------------------------------
	  // Load VTK library and print which library was not properly loaded
	  static {
	    if (!vtkNativeLibrary.LoadAllNativeLibraries()) {
	      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) {
	        if (!lib.IsLoaded()) {
	          System.out.println(lib.GetLibraryName() + " not loaded");
	        }
	      }
	    }
	    vtkNativeLibrary.DisableOutputWindow(null);
	  }
	  // -----------------------------------------------------------------
	  
	  
	  public static void main(String s[]) {
		  
		  
			   vtkNamedColors Color = new vtkNamedColors(); 
			  
			   //For Text Actor Color
			   double TextActorColor[] = new double[4];
			   //For Renderer Background Color
			   double BgColor[] = new double[4];
			   
			   
			   //Change Color Name to Use your own Color for Change Actor Color
			   Color.GetColor("IndianRed",TextActorColor);
			   //Change Color Name to Use your own Color for Renderer Background
			   Color.GetColor("LightGrey",BgColor);
			
			   
			   // Create the axes and the associated mapper and actor.
			   vtkAxes Axes = new vtkAxes();
			   Axes.SetOrigin(0, 0, 0);
			   
			   vtkPolyDataMapper AxesMapper = new vtkPolyDataMapper();
			   AxesMapper.SetInputConnection(Axes.GetOutputPort());
			   
			   vtkActor AxesActor = new vtkActor();
			   AxesActor.SetMapper(AxesMapper);
			  
			  // Create the 3D text and the associated mapper and follower (a type of actor).  
			  //Position the text so it is displayed over the origin of the axes.
			  vtkVectorText aText = new vtkVectorText();
			  aText.SetText("Origin");
			  
			  
			  vtkPolyDataMapper TextMapper = new vtkPolyDataMapper();
			  TextMapper.SetInputConnection(aText.GetOutputPort());
			  
			  
			  vtkFollower TextActor = new vtkFollower();
			  TextActor.SetMapper(TextMapper);
			  TextActor.SetScale(0.2, 0.2, 0.2);
			  TextActor.AddPosition(0, -0.1, 0);
			  TextActor.GetProperty().SetColor(TextActorColor);
			  
		  
			  // Create the renderer, render window and interactor.
			  vtkRenderer ren = new vtkRenderer();
			  vtkRenderWindow renWin = new vtkRenderWindow();
			  renWin.AddRenderer(ren);
			  vtkRenderWindowInteractor iren = new vtkRenderWindowInteractor();
			  iren.SetRenderWindow(renWin);
	        
			  // Visualise the arrow
			  ren.AddActor(TextActor);
			  ren.AddActor(AxesActor);
			  ren.SetBackground(BgColor);
			  ren.ResetCamera();
			  ren.ResetCameraClippingRange();
			  ren.GetActiveCamera().Zoom(1.6);

			  renWin.SetSize(300, 300);
			  renWin.Render();
		
			  iren.Initialize();
			  iren.Start();
	    
	  }
	} 

File: src/Python/Annotation/TextOrigin.py
#!/usr/bin/env python

import vtk

def main():
    colors = vtk.vtkNamedColors()
    
    # Create the axes and the associated mapper and actor.
    axes = vtk.vtkAxes()
    axes.SetOrigin(0, 0, 0)
    axesMapper = vtk.vtkPolyDataMapper()
    axesMapper.SetInputConnection(axes.GetOutputPort())
    axesActor = vtk.vtkActor()
    axesActor.SetMapper(axesMapper)
    
    # Create the 3D text and the associated mapper and follower (a type of actor).  Position the text so it is displayed over the origin of the axes.
    atext = vtk.vtkVectorText()
    atext.SetText("Origin")
    textMapper = vtk.vtkPolyDataMapper()
    textMapper.SetInputConnection(atext.GetOutputPort())
    textActor = vtk.vtkFollower()
    textActor.SetMapper(textMapper)
    textActor.SetScale(0.2, 0.2, 0.2)
    textActor.AddPosition(0, -0.1, 0)
    
    # Create the Renderer, RenderWindow, and RenderWindowInteractor.
    renderer = vtk.vtkRenderer()
    renderWindow = vtk.vtkRenderWindow()
    renderWindow.AddRenderer(renderer)
    renderWindow.SetSize(640, 480)
    
    interactor = vtk.vtkRenderWindowInteractor()
    interactor.SetRenderWindow(renderWindow)
    
    style = vtk.vtkInteractorStyleTrackballCamera()
    interactor.SetInteractorStyle( style )
    
    # Add the actors to the renderer.
    renderer.AddActor(axesActor)
    renderer.AddActor(textActor)
    
    renderer.SetBackground(colors.GetColor3d("Silver"))
    
    # Zoom in closer.
    renderer.ResetCamera()
    renderer.GetActiveCamera().Zoom(1.6)
    
    renderer.SetBackground(colors.GetColor3d("Silver"))
    
    # Reset the clipping range of the camera; set the camera of the follower; render.
    renderer.ResetCameraClippingRange()
    textActor.SetCamera(renderer.GetActiveCamera())
    
    interactor.Initialize()
    renderWindow.Render()
    interactor.Start()
	
if __name__ == '__main__':
    main()
